js2r-functions.el

```diff
@@ -298,12 +298,16 @@
 (defun js2r-extract-method (name)
   "Extract a method from the closest statement expression from the point."
   (interactive "sName of new method: ")
-  (js2r--extract-fn
-   name
-   (lambda ()
-       (goto-char (js2-node-abs-pos (js2r--closest #'js2-object-prop-node-p))))
-   "this.%s(%s);"
-   "%s: function (%s) {\n%s\n},\n\n"))
+  (let ((cls (js2r--closest #'js2-class-node-p)))
+    (js2r--extract-fn
+     name
+     (if cls (lambda ())
```

Comment:
`js2r--extract-fn` could be improved to take `nil` as `goto-position` function, instead of using an empty lambda.

Then your code could be simplified like this:

```lisp
(js2r--extract-fn
  name
  (when cls (goto-char ...))
```